File: miniprogram/pages/map/map.js
/**
 * 地图导览页面逻辑
 * 六堡茶AI智能助手
 */

const api = require('../../utils/api.js');

Page({
  data: {
    provinces: [],
    currentProvince: 'guangxi',
    currentProvinceName: '广西',
    centerLatitude: 23.5985,
    centerLongitude: 111.4228,
    scale: 12,
    markers: [],
    selectedPlace: null,
    showInfoCard: false,
    places: [],
    isLoading: false,
    userLatitude: null,
    userLongitude: null,
    hasLocationAuth: false
  },

  onLoad: function(options) {
    this.loadProvinces();
    this.checkLocationAuth();
    this.getUserLocation();
  },

  onShow: function() {
    if (typeof this.getTabBar === 'function' && this.getTabBar()) {
      this.getTabBar().setData({ selected: 2 });
    }
  },

  /**
   * 加载省份列表
   */
  loadProvinces: function() {
    const BASE_URL = api.BASE_URL || 'http://172.18.106.153:5000';
    
    const defaultProvinces = [
      { id: 'guangxi', name: '广西' },
      { id: 'yunnan', name: '云南' },
      { id: 'fujian', name: '福建' },
      { id: 'zhejiang', name: '浙江' },
      { id: 'anhui', name: '安徽' },
      { id: 'sichuan', name: '四川' },
      { id: 'guizhou', name: '贵州' },
      { id: 'hunan', name: '湖南' },
      { id: 'jiangsu', name: '江苏' },
      { id: 'henan', name: '河南' }
    ];
    
    this.setData({ 
      provinces: defaultProvinces,
      currentProvince: 'guangxi',
      currentProvinceName: '广西'
    });
    
    this.loadGardensByProvince('guangxi');
    
    wx.request({
      url: `${BASE_URL}/api/provinces`,
      method: 'GET',
      success: (res) => {
        if (res.data && res.data.success && res.data.data.length > 0) {
          this.setData({ provinces: res.data.data });
        }
      },
      fail: (err) => {
        console.error('加载省份失败，使用默认数据');
      }
    });
  },

  /**
   * 根据省份加载茶园数据
   */
  loadGardensByProvince: function(provinceId) {
    const BASE_URL = api.BASE_URL || 'http://172.18.106.153:5000';
    
    this.setData({ isLoading: true });
    
    wx.request({
      url: `${BASE_URL}/api/gardens/${provinceId}`,
      method: 'GET',
      success: (res) => {
        console.log('茶园数据返回:', res.data);
        if (res.data && res.data.success) {
          const data = res.data.data;
          const places = data.gardens.map(garden => ({
            ...garden,
            distance: garden.distance || '0km'
          }));
          
          console.log('解析后的地点数据:', places);
          
          this.setData({
            places: places,
            centerLatitude: data.center.latitude,
            centerLongitude: data.center.longitude,
            currentProvinceName: data.province,
            isLoading: false
          });
          
          this.initMarkers();
          this.calculateDistances();
        }
      },
      fail: (err) => {
        console.error('加载茶园数据失败', err);
        this.setData({ isLoading: false });
        wx.showToast({ title: '加载失败', icon: 'error' });
      }
    });
  },

  /**
   * 选择省份
   */
  selectProvince: function(e) {
    const id = e.currentTarget.dataset.id;
    const province = this.data.provinces.find(p => p.id === id);
    
    if (province) {
      this.setData({
        currentProvince: id,
        currentProvinceName: province.name,
        showInfoCard: false,
        selectedPlace: null
      });
      
      this.loadGardensByProvince(id);
    }
  },

  /**
   * 初始化地图标记点
   */
  initMarkers: function() {
    const places = this.data.places || [];
    const markers = places.map((place, index) => ({
      id: place.id,
      latitude: place.latitude,
      longitude: place.longitude,
      width: 40,
      height: 40,
      iconPath: '/miniprogram/images/marker.png',
      callout: {
        content: place.name,
        color: '#3E2723',
        fontSize: 14,
        borderRadius: 10,
        padding: 10,
        bgColor: '#FFFFFF',
        display: 'ALWAYS'
      }
    }));
    
    // 添加用户位置标记
    if (this.data.userLatitude && this.data.userLongitude) {
      markers.push({
        id: 999,
        latitude: this.data.userLatitude,
        longitude: this.data.userLongitude,
        width: 30,
        height: 30,
        callout: {
          content: '📍 我的位置',
          color: '#2E7D32',
          fontSize: 12,
          borderRadius: 8,
          padding: 6,
          bgColor: '#FFFFFF',
          display: 'ALWAYS'
        }
      });
    }
    
    this.setData({ markers });
  },

  /**
   * 检查位置权限
   */
  checkLocationAuth: function() {
    wx.getSetting({
      success: (res) => {
        if (res.authSetting['scope.userLocation']) {
          this.setData({ hasLocationAuth: true });
        }
      }
    });
  },

  /**
   * 获取用户位置
   */
  getUserLocation: function() {
    this.setData({ isLoading: true });
    
    wx.getLocation({
      type: 'gcj02',
      success: (res) => {
        console.log('用户位置', res);
        this.setData({
          userLatitude: res.latitude,
          userLongitude: res.longitude,
          hasLocationAuth: true,
          isLoading: false
        });
        this.calculateDistances();
        this.addUserMarker(res.latitude, res.longitude);
      },
      fail: (err) => {
        console.log('获取位置失败', err);
        this.setData({ isLoading: false });
        
        if (err.errMsg.indexOf('auth deny') > -1) {
          wx.showModal({
            title: '位置权限',
            content: '需要获取您的位置信息，以便提供附近茶园导航服务',
            confirmText: '去设置',
            success: (res) => {
              if (res.confirm) {
                wx.openSetting();
              }
            }
          });
        } else {
          wx.showToast({ title: '获取位置失败', icon: 'none' });
        }
      }
    });
  },

  /**
   * 添加用户位置标记
   */
  addUserMarker: function(latitude, longitude) {
    const userMarker = {
      id: 999,
      latitude: latitude,
      longitude: longitude,
      width: 30,
      height: 30,
      callout: {
        content: '📍 我的位置',
        color: '#2E7D32',
        fontSize: 12,
        borderRadius: 8,
        padding: 6,
        bgColor: '#FFFFFF',
        display: 'ALWAYS'
      }
    };
    
    const markers = [...this.data.markers, userMarker];
    this.setData({ markers });
  },

  /**
   * 计算各地点距离
   */
  calculateDistances: function() {
    if (!this.data.userLatitude) return;
    
    const places = this.data.places.map(place => {
      const distance = this.getDistance(
        this.data.userLatitude,
        this.data.userLongitude,
        place.latitude,
        place.longitude
      );
      return {
        ...place,
        distance: distance
      };
    });
    
    places.sort((a, b) => {
      const distA = parseFloat(a.distance) || Infinity;
      const distB = parseFloat(b.distance) || Infinity;
      return distA - distB;
    });
    
    this.setData({ places });
  },

  /**
   * 计算两点间距离（Haversine公式）
   */
  getDistance: function(lat1, lng1, lat2, lng2) {
    const R = 6371;
    const dLat = this.deg2rad(lat2 - lat1);
    const dLng = this.deg2rad(lng2 - lng1);
    const a = 
      Math.sin(dLat / 2) * Math.sin(dLat / 2) +
      Math.cos(this.deg2rad(lat1)) * Math.cos(this.deg2rad(lat2)) *
      Math.sin(dLng / 2) * Math.sin(dLng / 2);
    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
    const distance = R * c;
    
    if (distance < 1) {
      return Math.round(distance * 1000) + 'm';
    }
    return distance.toFixed(1) + 'km';
  },

  /**
   * 度数转弧度
   */
  deg2rad: function(deg) {
    return deg * (Math.PI / 180);
  },

  /**
   * 点击标记点
   */
  onMarkerTap: function(e) {
    const markerId = e.detail.markerId;
    if (markerId === 999) return;
    
    const place = this.data.places.find(p => p.id === markerId);
    if (place) {
      this.showPlaceInfo(place);
    }
  },

  /**
   * 点击地图其他区域
   */
  onMapTap: function() {
    this.setData({
      showInfoCard: false,
      selectedPlace: null
    });
  },

  /**
   * 显示地点信息卡
   */
  showPlaceInfo: function(place) {
    this.setData({
      selectedPlace: place,
      showInfoCard: true
    });
  },

  /**
   * 隐藏信息卡
   */
  hideInfoCard: function() {
    this.setData({
      showInfoCard: false,
      selectedPlace: null
    });
  },

  /**
   * 开始导航
   */
  startNavigation: function() {
    const place = this.data.selectedPlace;
    if (!place) return;
    
    wx.openLocation({
      latitude: place.latitude,
      longitude: place.longitude,
      name: place.name,
      address: place.address,
      scale: 16
    });
  },

  /**
   * 查看详情
   */
  viewDetail: function() {
    const place = this.data.selectedPlace;
    if (!place) return;
    
    wx.showModal({
      title: place.name,
      content: `${place.description}\n\n📍 地址：${place.address}\n🕐 营业时间：${place.openingHours}\n🏷️ 特色：${place.features.join('、')}`,
      confirmText: '开始导航',
      cancelText: '关闭',
      success: (res) => {
        if (res.confirm) {
          this.startNavigation();
        }
      }
    });
  },

  /**
   * 切换地点（列表点击）
   */
  selectPlace: function(e) {
    const placeId = e.currentTarget.dataset.id;
    const place = this.data.places.find(p => p.id === placeId);
    
    if (place) {
      this.setData({
        centerLatitude: place.latitude,
        centerLongitude: place.longitude,
        scale: 14
      });
      this.showPlaceInfo(place);
    }
  },

  /**
   * 重新定位到用户位置
   */
  relocateToUser: function() {
    if (this.data.userLatitude && this.data.userLongitude) {
      this.setData({
        centerLatitude: this.data.userLatitude,
        centerLongitude: this.data.userLongitude,
        scale: 14,
        showInfoCard: false,
        selectedPlace: null
      });
      wx.showToast({ title: '已定位到您的位置', icon: 'success', duration: 1500 });
    } else {
      this.getUserLocation();
    }
  },

  onShareAppMessage: function() {
    return {
      title: '六堡茶茶园地图导览',
      path: '/pages/map/map'
    };
  },

  navigateBack: function() {
    wx.navigateBack();
  }
});

File: backend/map.py
# 默认数据（梧州市六堡茶核心产区）
DEFAULT_GARDENS = {
    'province': '广西',
    'center': {'latitude': 23.5985, 'longitude': 111.4228},
    'gardens': [
        {
            'id': 1,
            'name': '六堡镇茶园',
            'latitude': 23.5985,
            'longitude': 111.4228,
            'icon': '🌿',
            'distance': '0km',
            'description': '六堡茶核心产区，拥有千年种茶历史，是六堡茶的发源地。',
            'features': ['千年古树', '传统工艺', '槟榔香'],
            'address': '广西梧州市苍梧县六堡镇',
            'openingHours': '08:00-18:00',
            'phone': '0774-8888888'
        },
        {
            'id': 2,
            'name': '苍梧生态茶园',
            'latitude': 23.6123,
            'longitude': 111.4356,
            'icon': '🌱',
            'distance': '2.5km',
            'description': '现代化生态茶园，采用有机种植方式，年产优质六堡茶50吨。',
            'features': ['有机种植', '生态示范', '观光体验'],
            'address': '广西梧州市苍梧县六堡镇苍梧茶园',
            'openingHours': '08:00-18:00',
            'phone': '0774-1234567'
        },
        {
            'id': 3,
            'name': '六堡茶博园',
            'latitude': 23.5856,
            'longitude': 111.4089,
            'icon': '🏛️',
            'distance': '3.8km',
            'description': '六堡茶文化博物馆和体验中心，展示六堡茶历史文化和制作工艺。',
            'features': ['文化展示', '制茶体验', '品茗休闲'],
            'address': '广西梧州市苍梧县六堡镇茶博园',
            'openingHours': '09:00-17:00',
            'phone': '0774-7654321'
        },
        {
            'id': 4,
            'name': '黑石山茶园',
            'latitude': 23.6200,
            'longitude': 111.4500,
            'icon': '⛰️',
            'distance': '5.2km',
            'description': '六堡茶核心产区之一的黑石山，以产高品质六堡茶闻名。',
            'features': ['高山茶', '传统工艺', '精品六堡'],
            'address': '广西梧州市苍梧县六堡镇黑石山',
            'openingHours': '08:00-17:00',
            'phone': '0774-6666666'
        },
        {
            'id': 5,
            'name': '恭州村茶园',
            'latitude': 23.5750,
            'longitude': 111.4000,
            'icon': '🏡',
            'distance': '4.5km',
            'description': '六堡镇"四柳"之一的恭州村，以茶气足、口感醇厚著称。',
            'features': ['恭州茶', '茶气足', '传统工艺'],
            'address': '广西梧州市苍梧县六堡镇恭州村',
            'openingHours': '08:00-18:00',
            'phone': '0774-5555555'
        },
        {
            'id': 6,
            'name': '蚕村茶园',
            'latitude': 23.5900,
            'longitude': 111.4150,
            'icon': '🌾',
            'distance': '3.2km',
            'description': '六堡镇"四柳"之一的蚕村，出产的茶叶口感醇和。',
            'features': ['蚕村茶', '口感醇和', '传统工艺'],
            'address': '广西梧州市苍梧县六堡镇蚕村',
            'openingHours': '08:00-17:30',
            'phone': '0774-4444444'
        },
        {
            'id': 7,
            'name': '大宁村茶园',
            'latitude': 23.6050,
            'longitude': 111.4280,
            'icon': '🌳',
            'distance': '1.8km',
            'description': '六堡镇"四柳"之一的大宁村，茶汤色红亮，品质优良。',
            'features': ['大宁茶', '汤色红亮', '品质优良'],
            'address': '广西梧州市苍梧县六堡镇大宁村',
            'openingHours': '08:00-18:00',
            'phone': '0774-3333333'
        }
    ]
}

# 各省茶园数据
TEA_GARDENS = {
    'guangxi': DEFAULT_GARDENS,
    'yunnan': {
        'province': '云南',
        'center': {'latitude': 22.9876, 'longitude': 101.0123},
        'gardens': [
            {'id': 8, 'name': '普洱古茶园', 'latitude': 22.9876, 'longitude': 101.0123, 'icon': '🌳', 'distance': '0km', 'description': '世界文化遗产地，拥有千年古茶树群。', 'features': ['千年古树', '普洱茶'], 'address': '云南普洱市宁洱县', 'openingHours': '08:30-18:30', 'phone': '0879-6666666'},
            {'id': 9, 'name': '景迈山茶园', 'latitude': 22.1956, 'longitude': 100.0234, 'icon': '🌄', 'distance': '8.2km', 'description': '全球重要农业文化遗产，景迈山古茶林。', 'features': ['世界遗产', '古茶林'], 'address': '云南普洱市澜沧县', 'openingHours': '09:00-17:00', 'phone': '0879-5555555'}
        ]
    },
    'fujian': {
        'province': '福建',
        'center': {'latitude': 27.7567, 'longitude': 117.9857},
        'gardens': [
            {'id': 10, 'name': '武夷山岩茶基地', 'latitude': 27.7567, 'longitude': 117.9857, 'icon': '⛰️', 'distance': '0km', 'description': '世界文化与自然双重遗产，大红袍发源地。', 'features': ['世界遗产', '岩韵', '大红袍'], 'address': '福建武夷山市', 'openingHours': '08:00-18:00', 'phone': '0599-8888888'},
            {'id': 11, 'name': '铁观音茶山', 'latitude': 25.1234, 'longitude': 117.8765, 'icon': '🍃', 'distance': '12.3km', 'description': '安溪铁观音核心产区，中国十大名茶之一。', 'features': ['铁观音', '兰花香'], 'address': '福建泉州市安溪县', 'openingHours': '08:30-17:30', 'phone': '0595-6666666'}
        ]
    },
    'zhejiang': {
        'province': '浙江',
        'center': {'latitude': 30.1245, 'longitude': 120.1234},
        'gardens': [
            {'id': 12, 'name': '龙井茶园', 'latitude': 30.1245, 'longitude': 120.1234, 'icon': '🍵', 'distance': '0km', 'description': '西湖龙井核心产区，中国绿茶代表。', 'features': ['西湖龙井', '狮峰山'], 'address': '浙江杭州市西湖区', 'openingHours': '08:30-17:30', 'phone': '0571-8888888'},
            {'id': 13, 'name': '安吉白茶园', 'latitude': 30.5678, 'longitude': 119.5678, 'icon': '🌿', 'distance': '8.5km', 'description': '安吉白茶核心产区，氨基酸含量高。', 'features': ['安吉白茶', '氨基酸'], 'address': '浙江湖州市安吉县', 'openingHours': '08:00-17:00', 'phone': '0572-7777777'}
        ]
    },
    'anhui': {
        'province': '安徽',
        'center': {'latitude': 30.4567, 'longitude': 118.0234},
        'gardens': [
            {'id': 14, 'name': '黄山毛峰茶园', 'latitude': 30.4567, 'longitude': 118.0234, 'icon': '🏔️', 'distance': '0km', 'description': '黄山毛峰核心产区，中国十大名茶之一。', 'features': ['黄山毛峰', '黄山云雾'], 'address': '安徽黄山市徽州区', 'openingHours': '08:00-18:00', 'phone': '0559-8888888'},
            {'id': 15, 'name': '祁门红茶园', 'latitude': 29.8765, 'longitude': 117.6543, 'icon': '🌹', 'distance': '10.2km', 'description': '祁门红茶核心产区，世界三大高香红茶之一。', 'features': ['祁门红茶', '高香'], 'address': '安徽黄山市祁门县', 'openingHours': '08:30-17:30', 'phone': '0559-6666666'}
        ]
    },
    'sichuan': {
        'province': '四川',
        'center': {'latitude': 30.5678, 'longitude': 103.4567},
        'gardens': [
            {'id': 16, 'name': '蒙顶山茶园', 'latitude': 30.5678, 'longitude': 103.4567, 'icon': '⛰️', 'distance': '0km', 'description': '中国茶文化发源地之一，蒙顶山茶千年贡茶。', 'features': ['茶文化发源', '蒙顶甘露'], 'address': '四川雅安市名山区', 'openingHours': '08:00-18:00', 'phone': '0835-8888888'},
            {'id': 17, 'name': '峨眉山茶园', 'latitude': 29.6543, 'longitude': 103.3210, 'icon': '🏔️', 'distance': '15.3km', 'description': '峨眉山茶区，世界文化与自然遗产。', 'features': ['峨眉山', '竹叶青'], 'address': '四川乐山市峨眉山市', 'openingHours': '08:30-17:30', 'phone': '0833-7777777'}
        ]
    },
    'guizhou': {
        'province': '贵州',
        'center': {'latitude': 26.5678, 'longitude': 106.1234},
        'gardens': [
            {'id': 18, 'name': '都匀毛尖茶园', 'latitude': 26.5678, 'longitude': 106.1234, 'icon': '🍃', 'distance': '0km', 'description': '都匀毛尖核心产区，中国十大名茶之一。', 'features': ['都匀毛尖', '贵州绿茶'], 'address': '贵州黔南州都匀市', 'openingHours': '08:00-18:00', 'phone': '0854-8888888'}
        ]
    },
    'hunan': {
        'province': '湖南',
        'center': {'latitude': 28.1234, 'longitude': 112.5678},
        'gardens': [
            {'id': 19, 'name': '君山银针茶园', 'latitude': 28.1234, 'longitude': 112.5678, 'icon': '🌱', 'distance': '0km', 'description': '君山银针核心产区，中国十大名茶之一。', 'features': ['君山银针', '黄茶'], 'address': '湖南岳阳市君山区', 'openingHours': '08:30-17:30', 'phone': '0730-8888888'}
        ]
    },
    'jiangsu': {
        'province': '江苏',
        'center': {'latitude': 32.0123, 'longitude': 120.5678},
        'gardens': [
            {'id': 20, 'name': '碧螺春茶园', 'latitude': 32.0123, 'longitude': 120.5678, 'icon': '🍃', 'distance': '0km', 'description': '洞庭碧螺春核心产区，中国十大名茶之一。', 'features': ['碧螺春', '洞庭山'], 'address': '江苏苏州市吴中区太湖', 'openingHours': '08:00-18:00', 'phone': '0512-8888888'}
        ]
    },
    'henan': {
        'province': '河南',
        'center': {'latitude': 33.4567, 'longitude': 113.1234},
        'gardens': [
            {'id': 21, 'name': '信阳毛尖茶园', 'latitude': 33.4567, 'longitude': 113.1234, 'icon': '🌿', 'distance': '0km', 'description': '信阳毛尖核心产区，中国十大名茶之一。', 'features': ['信阳毛尖', '豫毛峰'], 'address': '河南信阳市浉河区', 'openingHours': '08:00-18:00', 'phone': '0376-8888888'}
        ]
    },
    # 新增省份
    'hubei': {
        'province': '湖北',
        'center': {'latitude': 30.2345, 'longitude': 110.0123},
        'gardens': [
            {'id': 22, 'name': '恩施玉露茶园', 'latitude': 30.2345, 'longitude': 110.0123, 'icon': '🍵', 'distance': '0km', 'description': '保留唐代蒸青工艺，湖北历史名茶。', 'features': ['蒸青绿茶', '恩施富硒'], 'address': '湖北恩施市芭蕉乡', 'openingHours': '08:00-18:00', 'phone': '0718-8888888'},
            {'id': 23, 'name': '赤壁青砖茶基地', 'latitude': 29.7654, 'longitude': 113.6789, 'icon': '🧱', 'distance': '11.6km', 'description': '赤壁青砖茶原产地，传统紧压黑茶。', 'features': ['青砖茶', '边销茶'], 'address': '湖北咸宁市赤壁市', 'openingHours': '08:30-17:30', 'phone': '0715-7777777'}
        ]
    },
    'guangdong': {
        'province': '广东',
        'center': {'latitude': 23.6789, 'longitude': 113.5678},
        'gardens': [
            {'id': 24, 'name': '凤凰山单丛茶园', 'latitude': 23.6789, 'longitude': 113.5678, 'icon': '⛰️', 'distance': '0km', 'description': '凤凰单丛核心产区，香型丰富多变。', 'features': ['凤凰单丛', '高山乌龙'], 'address': '广东潮州市潮安区凤凰镇', 'openingHours': '08:00-18:00', 'phone': '0768-8888888'},
            {'id': 25, 'name': '英德红茶园', 'latitude': 24.1234, 'longitude': 113.2345, 'icon': '🍂', 'distance': '22.3km', 'description': '英红九号原产地，广东知名红茶。', 'features': ['英红九号', '蜜香红茶'], 'address': '广东清远英德市', 'openingHours': '08:30-17:30', 'phone': '0763-6666666'}
        ]
    },
    'jiangxi': {
        'province': '江西',
        'center': {'latitude': 28.5678, 'longitude': 115.6789},
        'gardens': [
            {'id': 26, 'name': '庐山云雾茶园', 'latitude': 28.5678, 'longitude': 115.6789, 'icon': '🏔️', 'distance': '0km', 'description': '庐山高山云雾茶，清香鲜爽。', 'features': ['高山云雾', '绿茶'], 'address': '江西九江市庐山市', 'openingHours': '08:00-18:00', 'phone': '0792-8888888'},
            {'id': 27, 'name': '遂川狗牯脑茶园', 'latitude': 26.3456, 'longitude': 114.5678, 'icon': '🌱', 'distance': '35.7km', 'description': '江西历史贡茶，滋味醇厚回甘。', 'features': ['狗牯脑', '名优绿茶'], 'address': '江西吉安市遂川县', 'openingHours': '08:30-17:00', 'phone': '0796-5555555'}
        ]
    },
    'shandong': {
        'province': '山东',
        'center': {'latitude': 36.4567, 'longitude': 120.8901},
        'gardens': [
            {'id': 28, 'name': '崂山绿茶园', 'latitude': 36.4567, 'longitude': 120.8901, 'icon': '🌿', 'distance': '0km', 'description': '北方高纬度茶园，茶汤鲜爽耐泡。', 'features': ['北方茶区', '崂山绿茶'], 'address': '山东青岛市崂山区', 'openingHours': '08:00-18:00', 'phone': '0532-8888888'}
        ]
    },
    'shanxi': {
        'province': '陕西',
        'center': {'latitude': 32.7890, 'longitude': 108.5678},
        'gardens': [
            {'id': 29, 'name': '紫阳富硒茶园', 'latitude': 32.7890, 'longitude': 108.5678, 'icon': '🍃', 'distance': '0km', 'description': '北方富硒茶产区，紫阳毛尖清香回甘。', 'features': ['富硒茶', '紫阳毛尖'], 'address': '陕西安康紫阳县', 'openingHours': '08:00-18:00', 'phone': '0915-8888888'}
        ]
    },
    'taiwan': {
        'province': '台湾',
        'center': {'latitude': 23.475537, 'longitude': 120.696485},
        'gardens': [
            {'id': 30, 'name': '阿里山高山茶园', 'latitude': 24.1234, 'longitude': 121.6789, 'icon': '☁️', 'distance': '0km', 'description': '阿里山高山乌龙，香气清雅绵长。', 'features': ['高山乌龙', '奶香果香'], 'address': '台湾嘉义阿里山乡', 'openingHours': '全天观光采摘时段08:00-17:00', 'phone': '05-2888888'}
        ]
    }
}

# 省份列表（同步新增全部省份）
PROVINCES = [
    {'id': 'guangxi', 'name': '广西'},
    {'id': 'yunnan', 'name': '云南'},
    {'id': 'fujian', 'name': '福建'},
    {'id': 'zhejiang', 'name': '浙江'},
    {'id': 'anhui', 'name': '安徽'},
    {'id': 'sichuan', 'name': '四川'},
    {'id': 'guizhou', 'name': '贵州'},
    {'id': 'hunan', 'name': '湖南'},
    {'id': 'jiangsu', 'name': '江苏'},
    {'id': 'henan', 'name': '河南'},
    {'id': 'hubei', 'name': '湖北'},
    {'id': 'guangdong', 'name': '广东'},
    {'id': 'jiangxi', 'name': '江西'},
    {'id': 'shandong', 'name': '山东'},
    {'id': 'shanxi', 'name': '陕西'},
    {'id': 'taiwan', 'name': '台湾'}
]

if __name__ == "__main__":
    # 简易测试代码
    total_garden = 0
    for p in TEA_GARDENS.values():
        total_garden += len(p["gardens"])
    print(f"全国收录茶园总数量：{total_garden} 处")
    print(f"覆盖省份总数：{len(PROVINCES)} 个")
    print("广西第一处茶园：", TEA_GARDENS["guangxi"]["gardens"][0]["name"])